File: Java/src/Subsets II .java
public class Solution {
    public List<List<Integer>> subsetsWithDup(int[] num) {
        Arrays.sort(num);
        List<List<Integer>> ret = new ArrayList<>();
        int length = num.length;
        if (length == 0) return ret;
        ret.add(new ArrayList<Integer>());
        ret.add(new ArrayList<Integer>());
        ret.get(1).add(num[0]);
        
        int lastRound = 1;
        
        for (int i = 1; i<length; i++){
            int size = ret.size();
            if(num[i] != num[i-1]) lastRound = size;
            for (int j = ret.size() - lastRound; j< size; j++){
                List<Integer> l = new ArrayList<Integer>(ret.get(j));
                l.add(num[i]);
                ret.add(l);
            }
        }
        
        return ret;
    }
}

File: Python/Subsets II .py
import copy
class Solution:
    # @param num, a list of integer
    # @return a list of lists of integer
    def subsetsWithDup(self, S):
        S.sort()
        ret = [[]]
        length = len(S)
        if length == 0:
            return ret
        last = [[S[0]]]
        ret = [[],[S[0]]]
        for i in range(1, length):
            if S[i] == S[i-1]:
                for l in last:
                    l.append(S[i])
                ret.extend(copy.deepcopy(last))
            else:
                last = copy.deepcopy(ret)
                for l in last:
                    l.append(S[i])
                ret.extend(copy.deepcopy(last))
        return ret
            
s = Solution()
print s.subsetsWithDup([1,1])